rosidl_generator_py/resource/_msg.py.em

```diff
@@ -428,28 +438,28 @@ if member.name in dict(inspect.getmembers(builtins)).keys():
     def @(member.name)(self, value):@(noqa_string)
-@[  if isinstance(member.type, AbstractNestedType) and isinstance(member.type.value_type, BasicType) and member.type.value_type.typename in SPECIAL_NESTED_BASIC_TYPES]@
-@[    if isinstance(member.type, Array)]@
-        if isinstance(value, numpy.ndarray):
-            assert value.dtype == @(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['dtype']), \
-                "The '@(member.name)' numpy.ndarray() must have the dtype of '@(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['dtype'])'"
-            assert value.size == @(member.type.size), \
-                "The '@(member.name)' numpy.ndarray() must have a size of @(member.type.size)"
-            self._@(member.name) = value
-            return
-@[    elif isinstance(member.type, AbstractSequence)]@
-        if isinstance(value, array.array):
-            assert value.typecode == '@(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['type_code'])', \
-                "The '@(member.name)' array.array() must have the type code of '@(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['type_code'])'"
-@[      if isinstance(member.type, BoundedSequence)]@
-            assert len(value) <= @(member.type.maximum_size), \
-                "The '@(member.name)' array.array() must have a size <= @(member.type.maximum_size)"
+        if self._check_fields or ros_python_check_fields:
+@[      if isinstance(member.type, AbstractNestedType) and isinstance(member.type.value_type, BasicType) and member.type.value_type.typename in SPECIAL_NESTED_BASIC_TYPES]@
+@[        if isinstance(member.type, Array)]@
+            if isinstance(value, numpy.ndarray):
+                assert value.dtype == @(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['dtype']), \
+                    "The '@(member.name)' numpy.ndarray() must have the dtype of '@(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['dtype'])'"
+                assert value.size == @(member.type.size), \
+                    "The '@(member.name)' numpy.ndarray() must have a size of @(member.type.size)"
+                self._@(member.name) = value
+                return
+@[        elif isinstance(member.type, AbstractSequence)]@
+            if isinstance(value, array.array):
+                assert value.typecode == '@(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['type_code'])', \
+                    "The '@(member.name)' array.array() must have the type code of '@(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['type_code'])'"
+@[          if isinstance(member.type, BoundedSequence)]@
+                assert len(value) <= @(member.type.maximum_size), \
+                    "The '@(member.name)' array.array() must have a size <= @(member.type.maximum_size)"
+@[          end if]@
+                self._@(member.name) = value
+                return
+@[        end if]@
```

Comment:
I'm not totally sure, but I think the `empy` if statements are now over-indented.  That is, they are all inside of the `@[for member in message.structure.members]@` loop starting at line 418, so I think they should only be indented by two spaces.

That will also make it easier to review this PR, as the only change will be from `if __debug__` to `if self._check_fields`.

Reply:
Yeah, the `empy` statements were over-indented. I already fixed it and I think it is easier to review that part as well.